ansible/roles/vm_set/files/mux_simulator.py

```diff
@@ -534,87 +534,87 @@ def clear_flap_counter(self):
 
 class Muxes(object):
 
-    def __init__(self):
+    def __init__(self, vm_set):
+        self.vm_set = vm_set
         self.muxes = {}
         for bridge in self._mux_bridges():
             bridge_fields = bridge.split('-')
-            vm_set = '-'.join(bridge_fields[1:-1])
             port_index = int(bridge_fields[-1])
             mux = Mux(vm_set, port_index)
             if mux.isvalid:
                 self.muxes[bridge] = mux
 
     def _mux_bridges(self):
-        return [intf for intf in os.listdir('/sys/class/net') if intf.startswith(MUX_BRIDGE_PREFIX)]
+        """Only collect bridges belong to self.vm_set
 
-    def filter_muxes(self, vm_set, port_index=None):
-        """Filter mux objects belong to specified vm_set.
+        Returns:
+            list: List of bridges belong to self.vm_set
         """
-        if port_index is not None:
-            bridge = adaptive_name(MUX_BRIDGE_TEMPLATE, vm_set, port_index)
-            return self.muxes[bridge]
-        else:
-            return [mux for bridge, mux in self.muxes.items() if vm_set in bridge]
+        pattern = '{}[^-]*-{}-[\d]+'.format(MUX_BRIDGE_PREFIX, self.vm_set)
```

Comment:
The regex pattern ```{}[^-]*-{}-[\d]+``` confused me. Is it something like ```muxy-vms-17-7-1```? Should it be ```{}-{}-[\d]+``` ?

Reply:
Because the max length of a bridge name is 16, we added `adaptive_name` to limit the bridge name length. As a consequence, the bridge name may start with `mbr-` or `mb-`. Here `MUX_BRIDGE_PREFIX` is `mb`. Simply using pattern `{}-{}-[\d]+` is not able to match bridge name starts with `mbr-`.

To ensure that the pattern can match both prefixes, we need to allow extra non `-` characters between `mb` and `-`. That's why I added `[^-]*` after the first `{}`.